server/auth/routes.py

```diff
@@ -317,6 +340,142 @@ def auditboard_passphrase(passphrase: str):
     )
 
 
+@auth.route("/tallyentry/<passphrase>", methods=["GET"])
+def tally_entry_passphrase(passphrase: str):
+    jurisdiction = Jurisdiction.query.filter_by(
+        tally_entry_passphrase=passphrase
+    ).one_or_none()
+    # TODO redirect to a nice error screen that explains they probably made a typo
+    if jurisdiction is None:
+        raise NotFound()
+
+    tally_entry_user = TallyEntryUser(
+        id=str(uuid.uuid4()), jurisdiction_id=jurisdiction.id,
+    )
+    db_session.add(tally_entry_user)
+    db_session.commit()
+
+    # We set the tally entry user in the session even though they haven't fully
+    # logged in yet. This allows the user to hit /api/me to retrieve
+    # jurisdiction name and login code to show on the login screens. The
+    # restrict_access decorator ensures that they can't do anything else until
+    # their login code is confirmed by the JA.
+    set_loggedin_user(session, UserType.TALLY_ENTRY, tally_entry_user.id)
+
+    return redirect("/tally-entry")
```

Comment:
This is the frontend route where tally entry content will live. I know it's confusingly close to the login link (`/tally-entry` vs `/tallyentry/<passphrase>`), but I thought this would be an easy way to avoid collisions with frontend routing/proxying. Though I didn't actually try out using overlapping URLs, so it might not be an issue. Dunno if anyone has experience with this.

Reply:
one trend I've seen is to have all APIs behind `/api/` to differentiate URLs that should show up in the browser bar from ones that are just JSON. But this seems fine with comments.

Reply:
Yeah, both of these routes are user-facing, but one has to hit the server while one hits the front-end routing, so I don't think the `/api/` approach will help here.